src/main/java/pl/debuguj/parkingspacessystem/dao/ParkingSpaceDaoMockImpl.java

```diff
@@ -0,0 +1,117 @@
+package pl.debuguj.parkingspacessystem.dao;
+
+import org.joda.time.DateTime;
+import org.springframework.stereotype.Service;
+import pl.debuguj.parkingspacessystem.domain.ParkingSpace;
+import pl.debuguj.parkingspacessystem.exceptions.CarRegisteredInSystemException;
+import pl.debuguj.parkingspacessystem.exceptions.IncorrectEndDateException;
+import pl.debuguj.parkingspacessystem.exceptions.ParkingSpaceNotFoundException;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Created by grzesiek on 07.10.17.
+ */
+@Service
+public class ParkingSpaceDaoMockImpl implements ParkingSpaceDao {
+
+    private static List<ParkingSpace> listParkingSpaces = new ArrayList<>();
+
+    @Override
+    public void create(final ParkingSpace parkingSpace) throws CarRegisteredInSystemException {
+
+        Optional<ParkingSpace> ps =
+                findByRegistrationNo(parkingSpace.getCarRegistrationNumber());
+        if(!ps.isPresent())
+            listParkingSpaces.add(parkingSpace);
+        else
+            throw new CarRegisteredInSystemException("Car is registered into system");
+    }
+
+    @Override
+    public Optional<ParkingSpace> findByRegistrationNo(final String registrationNo)  {
+        return listParkingSpaces
+                .stream()
+                .filter(ps -> Objects.equals(registrationNo, ps.getCarRegistrationNumber()))
+                .findFirst();
```

Comment:
a dlaczego `findFirst()`? Co się stanie jak dany pojazd będzie parkował kilkukrotnie?

Reply:
W tej chwili nie ma możliwości zaparkowania dwa razy samochodu o tej samej rejestracji. To błąd który zauważyłem dopiero dziś.